### PR TITLE
table(test): add unit tests for coverage of package table

### PR DESCRIPTION
**Pull request by jackysp**
Coverage of package table to 80%.
PTAL @shenli @zimulala

### CODE REVIEW

File: table/table_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2016 PingCAP, Inc.
```
Comment (shenli) [inline table/table_test.go]: 2016->2017

File: table/table_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/store/localstore"
+	"github.com/pingcap/tidb/store/localstore/goleveldb"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+var _ = Suite(&testSuite{})
```
Comment (shenli) [inline table/table_test.go]: We should use only one suit in a package.

Comment (jackysp) [inline table/table_test.go]: Done.

File: table/table_test.go (hunk 1 above)
Comment (jackysp) [inline table/table_test.go]: Done.

**Comment by jackysp** [comment]
Tests are simplified, PTAL @shenli

File: table/tables/memory_tables_test.go [hunk 3]
```diff
@@ -92,10 +111,13 @@ func (ts *testMemoryTableSuite) TestMemoryBasic(c *C) {
 	c.Assert(row[0].GetInt64(), Equals, int64(1))
 
 	_, err = tb.AddRecord(ctx, types.MakeDatums(1, "aba"))
-	c.Assert(err, IsNil)
+	c.Assert(err, NotNil)
```
Comment (shenli) [inline table/tables/memory_tables_test.go]: Why this is change?

File: table/tables/tables.go [hunk 4]
```diff
@@ -706,6 +706,9 @@ func (t *Table) RebaseAutoID(newBase int64, isSetStep bool) error {
 // Seek implements table.Table Seek interface.
 func (t *Table) Seek(ctx context.Context, h int64) (int64, bool, error) {
 	seekKey := tablecodec.EncodeRowKeyWithHandle(t.ID, h)
+	if ctx == nil {
```
Comment (shenli) [inline table/tables/tables.go]: ctx should never be nil.

File: table/tables/tables_test.go [hunk 5]
```diff
@@ -286,3 +304,24 @@ func (ts *testSuite) TestIterRecords(c *C) {
...
+	tbInfo := tb.Meta()
+	tbInfo.Columns[0].GeneratedExprString = "test"
+	tables.TableFromMeta(nil, tbInfo)
+	tbInfo.Columns[0].State = model.StateNone
+	tb, err = tables.TableFromMeta(nil, tbInfo)
+	c.Assert(tb, IsNil)
+	c.Assert(err, NotNil)
+	tbInfo.State = model.StateNone
+	tables.TableFromMeta(nil, tbInfo)
+	c.Assert(tb, IsNil)
```
Comment (shenli) [inline table/tables/tables_test.go]: tb and err are not changed after line 322. Why check them again?

File: table/tables/memory_tables_test.go (hunk 3 above)
Comment (jackysp) [inline table/tables/memory_tables_test.go]: The first column turns to be the primary key on line 68 and 70, so primary key conflicts here.

File: table/tables/tables.go (hunk 4 above)
Comment (jackysp) [inline table/tables/tables.go]: Ok, removed.

File: table/tables/tables_test.go (hunk 5 above)
Comment (jackysp) [inline table/tables/tables_test.go]: Add return values of line 324. The last two checks will work.

File: table/tables/memory_tables_test.go (hunk 3 above)
Comment (shenli) [inline table/tables/memory_tables_test.go]: Got it.

File: table/column_test.go [hunk 6]
```diff
@@ -189,7 +205,41 @@ func (s *testColumnSuite) TestGetZeroValue(c *C) {
...
+	err = CastValues(ctx, []types.Datum{types.NewDatum("test")}, []*Column{col}, true)
+	c.Assert(err, IsNil)
+
+	colInfoS := model.ColumnInfo{
+		FieldType: *types.NewFieldType(mysql.TypeString),
+		State:     model.StatePublic,
+	}
+	val, err = CastValue(ctx, types.NewDatum("test"), &colInfoS)
+	c.Assert(err, IsNil)
+	c.Assert(val, NotNil)
```
Comment (tiancaiamao) [inline table/column_test.go]: CastValue return types.Datum, so val would never be Nil, this line is useless.

**Pull request merged**